django-cloudlaunch/cloudlaunch/tests.py

```diff
@@ -140,3 +162,258 @@ def test_create_cloud_creds(self):
+        }
+        app_config = {
+            'bar': 3,
+            'baz': 4,
+            'config_cloudlaunch': {
+                'instance_user_data': "userdata"
+            }
+        }
+        app_version_cloud_config = \
+            ApplicationVersionCloudConfig.objects.create(
+                application_version=application_version,
+                cloud=target_cloud,
+                image=ubuntu_image,
+                default_launch_config=json.dumps(default_launch_config)
+            )
+        celery_id = str(uuid.uuid4())
+        launch_appliance_task.return_value = AsyncResult(celery_id)
+        url = reverse('deployments-list')
+        response = self.client.post(url, {
+            'name': 'test-deployment',
+            'application': application_version.application.slug,
+            'application_version': '16.04',
+            'target_cloud': target_cloud.slug,
+            'config_app': json.dumps(app_config),
+        })
+        self.assertEqual(response.status_code, 201)
+        launch_appliance_task.assert_called_with(
+            'test-deployment',
+            app_version_cloud_config.id,
+            credentials.as_dict(),
+            {
+                'foo': 1,  # default from default_launch_config
+                'bar': 3,  # config_app overrides default_launch_config
+                'baz': 4,  # added by config_app
+                'config_cloudlaunch': {
+                    'instance_user_data': "userdata"
+                }
+            },
+            'userdata'
+        )
+        app_deployment = ApplicationDeployment.objects.get()
+        launch_task = ApplicationDeploymentTask.objects.get(
+                action=ApplicationDeploymentTask.LAUNCH,
+                deployment=app_deployment, celery_id=celery_id)
+        self.assertIsNotNone(launch_task)
+
+
+class ApplicationDeploymentTaskTests(BaseAuthenticatedAPITestCase):
+
+    def _create_test_deployment(self, user):
+        application = Application.objects.create(
+            name="Ubuntu",
+            status=Application.LIVE,
+        )
+        application_version = ApplicationVersion.objects.create(
+            application=application,
+            version="1.0",
+        )
+        target_cloud = cb_models.AWS.objects.create(
+            name='Amazon US East 1 - N. Virginia',
+            kind='cloud',
+        )
+        user_profile = cb_models.UserProfile.objects.create(user=user)
+        credentials = cb_models.AWSCredentials.objects.create(
+            cloud=target_cloud,
+            access_key='access_key',
+            secret_key='secret_key',
+            user_profile=user_profile,
+        )
+        app_deployment = ApplicationDeployment.objects.create(
+            owner=user,
+            application_version=application_version,
+            target_cloud=target_cloud,
+            credentials=credentials
+        )
+        return app_deployment
+
+    @patch("cloudlaunch.tasks.health_check.delay")
```

Comment:
I was wondering - since there is a celery option to execute the task synchronously, should we allow it to execute, or just mock it out?

Reply:
I decided to mock it out because I just want to guard against regressions when refactoring the code in the serializers and views. For the refactor it doesn't much matter what happens in the task itself.